### PR TITLE
should throw exception while the SQL clause unsupported in encrypt rule

### PR DESCRIPTION
**Pull request by cheese8**
Refer: #14248 

Changes proposed in this pull request:
- throw exception while the SQL clause unsupported in encrypt rule

### CODE REVIEW

**Comment by cheese8** [comment]
ShardingSphereException: The SQL clause '>' is unsupported in encrypt rule. should be thrown as below SQL:

LogicSQL:
select t_account.certificate_number, t_account.amount, t_account_bak.certificate_number from t_account, t_account_bak where t_account.certificate_number = t_account_bak.certificate_number and t_account.amount > ?

ActualSQL:
select t_account.cipher_certificate_number AS certificate_number, t_account.cipher_amount AS amount, t_account_bak.cipher_certificate_number AS certificate_number from t_account, t_account_bak where t_account.assisted_query_certificate_number = t_account_bak.assisted_query_certificate_number and t_account.cipher_amount > ?

File: shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/parameter/impl/EncryptPredicateParameterRewriter.java [hunk 1]
```diff
@@ -65,8 +66,17 @@ public void rewrite(final ParameterBuilder parameterBuilder, final SQLStatementC
         String tableName = encryptCondition.getTableName();
         String columnName = encryptCondition.getColumnName();
         return getEncryptRule().findAssistedQueryColumn(tableName, columnName).isPresent()
-                ? getEncryptRule().getEncryptAssistedQueryValues(schemaName, tableName, columnName, originalValues) 
-                        : getEncryptRule().getEncryptValues(schemaName, tableName, columnName, originalValues);
+                ? checkSortable(encryptCondition, getEncryptRule().getEncryptAssistedQueryValues(schemaName, tableName, columnName, originalValues)) 
+                        : checkSortable(encryptCondition, getEncryptRule().getEncryptValues(schemaName, tableName, columnName, originalValues));
+    }
+    
+    private List<Object> checkSortable(final EncryptCondition encryptCondition, final List<Object> values) {
```
Comment (strongduanmu) [inline shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/parameter/impl/EncryptPredicateParameterRewriter.java:73]: Hi @cheese8, can we remove return result of this method? Judging from the name of the checkSortable method, it seems that it only checks the values.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/condition/EncryptConditionEngine.java [hunk 2]
```diff
@@ -131,6 +128,17 @@
         return Optional.empty();
     }
     
+    private Optional<EncryptCondition> createEncryptCondition(final BinaryOperationExpression expression, final String tableName) {
+        String operator = expression.getOperator();
+        if (!LOGICAL_OPERATOR.contains(operator)) {
+            if (isSupportedOperator(operator)) {
+                return createCompareEncryptCondition(tableName, expression, operator, expression.getRight());
+            }
+            throw new ShardingSphereException(String.format("The SQL clause '%s' is unsupported in encrypt rule.", operator));
```
Comment (strongduanmu) [inline shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/condition/EncryptConditionEngine.java:137]: @cheese8 Please remove String.format here, ShardingSphereException has include this logic.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/condition/EncryptConditionEngine.java (hunk 2 above)
Comment (cheese8) [reply shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/condition/EncryptConditionEngine.java:137]: ok

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**